bundles/org.eclipse.swt/Eclipse SWT/gtk/org/eclipse/swt/widgets/ColorDialog.java

```diff
@@ -141,34 +158,46 @@ public RGB open () {
 		rgba.blue = (double) rgb.blue / 255;
 		rgba.alpha = 1;
 	}
-	GTK.gtk_color_chooser_set_rgba (handle, rgba);
-	if (rgbs != null) {
-		int colorsPerRow = 9;
-		long gdkRGBAS = OS.g_malloc(GdkRGBA.sizeof * rgbs.length);
-		rgba = new GdkRGBA ();
-		for (int i=0; i<rgbs.length; i++) {
-			RGB rgbS = rgbs[i];
-			if (rgbS != null) {
-				rgba.red = (double) rgbS.red / 255;
-				rgba.green = (double) rgbS.green / 255;
-				rgba.blue = (double) rgbS.blue / 255;
-				OS.memmove (gdkRGBAS + i * GdkRGBA.sizeof, rgba, GdkRGBA.sizeof);
-			}
+
+	if (GTK.GTK4) {
+		// TODO The ColorDialog API doesn't support setting a custom palette
+		// See https://gitlab.gnome.org/GNOME/gtk/-/issues/5936
+		// > The new GtkColorDialog api doesn't have this feature atm. We might
+		// > want to bring it back, but judging from application use of color
+		// > choosers, setting custom palettes is not very popular.
```

Comment:
This is probably really bad. But as mentioned in the linked issue, as well as the [documentation](https://docs.gtk.org/gtk4/class.ColorDialog.html), there is simply no way to customize the palette.

Meaning this is probably something that would first have to be implemented, but I don't think my GTK knowledge is sufficient to do so. :/